plugins/promela/hu.bme.mit.gamma.promela.verification/src/hu/bme/mit/gamma/promela/verification/PromelaArrayParser.xtend

```diff
@@ -10,26 +10,21 @@
  ********************************************************************************/
 package hu.bme.mit.gamma.promela.verification
 
-import hu.bme.mit.gamma.expression.model.EnumerationTypeDefinition
 import hu.bme.mit.gamma.expression.util.IndexHierarchy
+import hu.bme.mit.gamma.statechart.statechart.Region
 import hu.bme.mit.gamma.theta.verification.XstsArrayParser
-import hu.bme.mit.gamma.xsts.model.XSTS
 import hu.bme.mit.gamma.xsts.promela.transformation.util.Namings
 import hu.bme.mit.gamma.xsts.util.XstsActionUtil
+import java.util.HashMap
 import java.util.List
 import java.util.regex.Pattern
 
-import static extension hu.bme.mit.gamma.expression.derivedfeatures.ExpressionModelDerivedFeatures.*
 import static extension hu.bme.mit.gamma.xsts.promela.transformation.util.Namings.*
-import hu.bme.mit.gamma.expression.model.ArrayTypeDefinition
-import hu.bme.mit.gamma.expression.model.TypeDefinition
 
-class PromelaArrayParser implements XstsArrayParser{
+class PromelaArrayParser implements XstsArrayParser {
 	// Singleton
 	public static final PromelaArrayParser INSTANCE = new PromelaArrayParser
-	protected static final extension XstsActionUtil xstsActionUtil = XstsActionUtil.INSTANCE
-	
-	protected static XSTS xsts
+	protected static HashMap<String, String> enumMapping
```

Comment:
Always use interfaces in such cases: `protected static Map<String, String> enumMapping`